src/services/billing/cost-management/widgets/SpcProjectWiseUsageSummary.vue

```diff
@@ -1,46 +1,128 @@
+                                       :page-size="10"
+                                       :pagination-visible="false"
+                                       show-legend
+                                       class="table table-right"
+                                       @toggle-legend="handleToggleLegend"
+            />
+        </div>
+    </cost-dashboard-card-widget-layout>
 </template>
 
 <script lang="ts">
 import {
+    computed,
+    defineComponent,
     onUnmounted, reactive, toRefs, watch,
 } from '@vue/composition-api';
 import * as am4core from '@amcharts/amcharts4/core';
 import * as am4charts from '@amcharts/amcharts4/charts';
 import { PieChart } from '@amcharts/amcharts4/charts';
 import config from '@/lib/config';
 import { SpaceConnector } from '@spaceone/console-core-lib/space-connector';
-import { GRANULARITY } from '@/services/billing/cost-management/lib/config';
+import { GRANULARITY, GROUP_BY } from '@/services/billing/cost-management/lib/config';
 import ErrorHandler from '@/common/composables/error/errorHandler';
-import dayjs from 'dayjs';
-
+import CostDashboardCardWidgetLayout
+    from '@/services/billing/cost-management/widgets/modules/CostDashboardCardWidgetLayout.vue';
+import {
+    Legend, PieChartData, PieChartRawData, WidgetProps,
+} from '@/services/billing/cost-management/widgets/type';
+import { CURRENCY } from '@/store/modules/display/config';
+import CostDashboardDataTable from '@/services/billing/cost-management/widgets/modules/CostDashboardDataTable.vue';
+import { getPieChartDataAndLegends } from '@/services/billing/cost-management/widgets/lib/widget-data-helper';
+import { store } from '@/store';
+import { PChartLoader, PSkeleton } from '@spaceone/design-system';
+import { gray } from '@/styles/colors';
 
-const categoryKey = 'project_id';
-const valueName = 'usd_cost';
 
-const currentDay = dayjs().utc();
-const currentMonthPeriod = {
-    start: currentDay.startOf('month'),
-    end: currentDay,
-};
+const categoryKey = 'category';
+const valueName = 'value';
 
-interface ChartData {
+interface ChartData extends PieChartRawData {
     project_id: string;
     usd_cost: number;
 }
 
-export default {
+export default defineComponent<WidgetProps>({
     name: 'SpcProjectWiseUsageSummary',
-    components: {},
-
-    setup() {
+    components: {
+        CostDashboardCardWidgetLayout,
+        CostDashboardDataTable,
+        PChartLoader,
+        PSkeleton,
+    },
+    props: {
+        options: {
+            type: Object,
+            default: () => ({}),
+        },
+        period: {
+            type: Object,
+            default: () => ({}),
+        },
+        filters: {
+            type: Object,
+            default: () => ({}),
+        },
+        currency: {
+            type: String,
+            default: CURRENCY.USD,
+        },
+        currencyRates: {
+            type: Object,
+            default: () => ({}),
+        },
+    },
+    setup(props) {
```

Comment:
```suggestion
    setup(props: WidgetProps) {
```

타입정의가 되면 더 좋을 것 같아요 ㅎ

Reply:
밖에 
export default defineComponent<WidgetProps>
요렇게 해놨는데 또 한번 더 정의할까요???

Reply:
수정했습니당!~~!

Reply:
아하 해놓으셨군요! 못 봤어요!! 죄송함다 😅